x-pack/plugins/infra/public/components/shared/alerts/alerts_overview.tsx

```diff
@@ -0,0 +1,123 @@
+
+interface AlertsOverviewProps {
+  assetName: string;
+  dateRange: TimeRange;
+  onLoaded: (alertsCount?: AlertsCount) => void;
+  onRangeSelection?: HostsStateUpdater;
+}
+
+export const AlertsOverview = React.memo(
+  ({ assetName, dateRange, onLoaded, onRangeSelection }: AlertsOverviewProps) => {
+    const { services } = useKibanaContextForPlugin();
+    const [urlState, setUrlState] = useAssetDetailsUrlState();
+    const [alertStatus, setAlertStatus] = useState<AlertStatus>(
+      urlState?.alertStatus ?? ALERT_STATUS_ALL
+    );
+
+    const alertsEsQueryByStatus = useMemo(
+      () =>
+        createAlertsEsQuery({
+          dateRange,
+          hostNodeNames: [assetName],
+          status: alertStatus,
+        }),
+      [assetName, dateRange, alertStatus]
+    );
+
+    const alertsEsQuery = useMemo(
+      () =>
+        createAlertsEsQuery({
+          dateRange,
+          hostNodeNames: [assetName],
+          status: ALERT_STATUS_ALL,
+        }),
+      [assetName, dateRange]
+    );
+
+    const summaryTimeRange = useSummaryTimeRange(dateRange);
+
+    const {
+      charts,
+      triggersActionsUi: {
+        getAlertsStateTable: AlertsStateTable,
+        alertsTableConfigurationRegistry,
+        getAlertSummaryWidget: AlertSummaryWidget,
+      },
+    } = services;
+
+    const onBrushEnd: BrushEndListener = (brushEvent) => {
+      const { x } = brushEvent as XYBrushEvent;
+      if (x && onRangeSelection) {
+        const [start, end] = x;
+
+        const from = new Date(start).toISOString();
+        const to = new Date(end).toISOString();
+
+        onRangeSelection({ dateRange: { from, to } });
+      }
+    };
+
+    const chartProps = {
+      baseTheme: charts.theme.useChartsBaseTheme(),
+      onBrushEnd,
+    };
+
+    const handleAlertStatusChange = (id: AlertStatus) => {
+      setAlertStatus(id);
+      setUrlState({ alertStatus: id });
+    };
+
+    return (
+      <EuiFlexGroup direction="column" gutterSize="m" data-test-subj="hostsView-alerts">
+        <EuiFlexGroup justifyContent="flexStart" alignItems="center">
+          <EuiFlexItem grow={false}>
+            <AlertsStatusFilter onChange={handleAlertStatusChange} status={alertStatus} />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        <EuiFlexItem>
+          <AlertSummaryWidget
+            chartProps={chartProps}
+            featureIds={infraAlertFeatureIds}
+            filter={alertsEsQuery}
+            timeRange={summaryTimeRange}
+            onLoaded={onLoaded}
+            fullSize
+          />
+        </EuiFlexItem>
+        {
+          <EuiFlexItem>
+            <AlertsStateTable
+              alertsTableConfigurationRegistry={alertsTableConfigurationRegistry}
+              id={'assetDetailsAlertsTable'}
+              configurationId={AlertConsumers.OBSERVABILITY}
+              featureIds={[...infraAlertFeatureIds, AlertConsumers.OBSERVABILITY]}
```

Comment:
You can also move this outside the component. This will cause this will cause the `AlertsStateTable` to rerender every time too.

```suggestion
              featureIds={alertFeatureIds}
```

Reply:
good catch